prometheus_client/values.py

```diff
@@ -51,6 +52,12 @@ class MmapedValue(object):
 
         def __init__(self, typ, metric_name, name, labelnames, labelvalues, multiprocess_mode='', **kwargs):
             self._params = typ, metric_name, name, labelnames, labelvalues, multiprocess_mode
+            # This deprecation warning can go away in a few releases when removing the compatibility
+            if 'prometheus_multiproc_dir' in os.environ and 'PROMETHEUS_MULTIPROC_DIR' in os.environ:
```

Comment:
If I instantiate multiple metrics, I end up getting both warnings because this clause is triggered on the second metric after the capital environment variable is set. Is there a good way to not get two warnings in that case? Here is a quick example:
```python
from prometheus_client import generate_latest, CollectorRegistry, multiprocess, Gauge

registry = CollectorRegistry()
multiprocess.MultiProcessCollector(registry)

Gauge('metric_1', 'First metric', registry=registry)
Gauge('metric_2', 'Second metric', registry=registry)
```
Since they are just warnings, not a big deal, but it is a bit confusing.

One option would be to always warn if the lowercase is set, then only replace if the capitalized version is not set? Users with both set would only get the `You should declare the env variable in upper case PROMETHEUS_MULTIPROC_DIR` in that case.

Reply:
Yes, correct. I'd rather display the warning for the one which only defined the value in small letter. I'll remove the case for both.

Reply:
:+1: Sounds good, let me know once that is complete and I will test again.

Reply:
Have a look. I also change the message on the deprecation warning as it did not sound so much as one. I think the wording is much better now. Don't hesitate if you have any other proposal to do.